src/daft-local-execution/src/sources/in_memory.rs

```diff
@@ -18,7 +18,7 @@ impl InMemorySource {
 
 impl Source for InMemorySource {
     #[instrument(name = "InMemorySource::get_data", level = "info", skip(self))]
-    fn get_data(&self) -> SourceStream {
+    fn get_data(&self, in_order: bool) -> SourceStream {
```

Comment:
So I know this already existed within `MultiSender` but it's not very apparent from the code what this flag actually represents.

does it mean: 

A: the node does not change the output ordering
B: the data has been ordered
C: the node requires that the data is ordered

Reply:
Yeah.. still trying to figure out the best abstractions for the new executor, so apologies for the confusion here.

But to answer your question, it's supposed to indicate that the **parent node** requires that the data received is ordered.

Reply:
what about using some enums and additional trait methods just to make everything a bit more readable

```rs
enum Ordering {
  Unordered,
  Ordered,
  Unknown
}


trait Source {
    fn get_data(&self, ordering: &Ordering) -> Data;
}

pub trait IntermediateOperator {
    fn execute(&self, input: &Arc<MicroPartition>) -> DaftResult<Arc<MicroPartition>>;
    fn name(&self) -> &'static str;
    fn output_ordering(&self) -> Ordering
    fn required_input_ordering(&self) -> Ordering;
}


impl MultiSender {
  fn required_input_ordering(&self) -> Ordering;
}
```